libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -288,6 +297,32 @@ void AP_ICEngine::update(void)
         }
     }
 
+    // update the filtered RPM value
+    if (rpm.get_rpm(rpm_instance-1, rpm_value)) {
+        filtered_rpm_value =  _rpm_filter.apply(rpm_value, AP::scheduler().get_loop_period_s());
+    } else {
+        _rpm_filter.reset();
+    }
+
+    // check against redline RPM
+    if (redline_rpm > 0 && rpm.get_rpm(rpm_instance-1, rpm_value)) {
```

Comment:
fetching the on RPM 301 and 308, you could probably  just combine the filter apply in this if, I don't think there is any need to run the filter if the redline rpm param is 0.

Reply:
Fixed.